test/e2e/proza-message.test.js

```diff
@@ -8,76 +7,82 @@ describe('vl-proza-message', async () => {
-//        await message.blur();
-//        await assert.eventually.isFalse(message.isEditable());
-//        await assert.eventually.equal(message.getText(), 'decibel');
-//    });
 
-        // TODO variant met toaster foutmelding e2e introductie
+    it('als de gebruiker kan ik op de enter-toets drukken om mijn wijzigingen te bewaren', async () => {
+        const message = await vlProzaMessagePage.getMessageFirstDemo();
+        await message.edit();
+        await message.clear();
+        await message.type('decibel');
+        await message.confirm();
+        await assert.eventually.equal(message.getText(), 'decibel');
+        await message.edit();
+        await message.clear();
+        await message.type('foobar');
+        await message.confirm();
+    });
+
+    it('als gebruiker kan ik enter+shift invoeren om een line-break toe te voegen', async () => {
+        const message = await vlProzaMessagePage.getMessageFirstDemo();
+        await message.edit();
+        await message.clear();
+        await message.type('line');
+        await message.shiftEnter();
+        await message.append('break');
+        await message.confirm();
+        const text = await message.getText();
+        assert.isTrue(text.indexOf('\n') > 0);
+        await message.edit();
+        await message.clear();
+        await message.type('foobar');
+        await message.confirm();
+    });
+
+    it('als gebruiker kan ik WYSIWYG stijl toevoegen aan de tekst door tekst te selecteren', async () => {
+        const message = await vlProzaMessagePage.getMessageFirstDemo();
+        await message.edit();
+        await message.selectAllText();
+        await message.waitUntilWysiwygPresent();
+        await message.cancel();
+    });
+
+    it('als gebruiker kan ik buiten het tekstveld klikken om de bewerk modus te sluiten en mijn wijzigingen te bewaren', async () => {
+        const message = await vlProzaMessagePage.getMessageFirstDemo();
+        await assert.eventually.equal(message.getText(), 'foobar');
+        await message.edit();
+        await message.clear();
+        await message.type('decibel');
+        await assert.eventually.isTrue(message.isEditable());
+        await message.blur();
+        await assert.eventually.isFalse(message.isEditable());
+        await assert.eventually.equal(message.getText(), 'decibel');
+    });
+
+    it('als gebruiker zie ik een waarschuwing in een alert als het updaten misgaat', async () => {
+        const message = await vlProzaMessagePage.getMessageWithError();
+        await assert.eventually.equal(message.getText(), 'update zal fout gaan');
+        await message.edit();
+        await message.clear();
+        await message.type('decibel');
+        await message.confirm();
+        const alert = await vlProzaMessagePage.getAlert();
+        await assert.eventually.isTrue(alert.isWarning());
+        await assert.eventually.equal(alert.getTitle(), 'Technische storing');
```

Comment:
hier denk ik testen dat de alert in nen toaster zit, en dus page objects van beide gebruiken

Reply:
Enkel toaster lijkt nodig, die neemt de alert wel mee binnen